Userland/Libraries/LibCrypto/BigInt/SignedBigInteger.cpp

```diff
@@ -353,151 +353,32 @@ bool SignedBigInteger::operator>=(SignedBigInteger const& other) const
 
 SignedBigInteger::CompareResult SignedBigInteger::compare_to_double(double value) const
 {
-    VERIFY(!isnan(value));
-
-    if (isinf(value)) {
-        bool is_positive_infinity = __builtin_isinf_sign(value) > 0;
-        return is_positive_infinity ? CompareResult::DoubleGreaterThanBigInt : CompareResult::DoubleLessThanBigInt;
-    }
-
     bool bigint_is_negative = m_sign;
 
     bool value_is_negative = value < 0;
 
     if (value_is_negative != bigint_is_negative)
         return bigint_is_negative ? CompareResult::DoubleGreaterThanBigInt : CompareResult::DoubleLessThanBigInt;
 
-    // Value is zero, and from above the signs must be the same.
-    if (value == 0.0) {
-        VERIFY(!value_is_negative && !bigint_is_negative);
-        // Either we are also zero or value is certainly less than us.
-        return is_zero() ? CompareResult::DoubleEqualsBigInt : CompareResult::DoubleLessThanBigInt;
-    }
-
-    // If value is not zero but we are, then since the signs are the same value must be greater.
-    if (is_zero())
-        return CompareResult::DoubleGreaterThanBigInt;
-
-    constexpr u64 mantissa_size = 52;
-    constexpr u64 exponent_size = 11;
-    constexpr auto exponent_bias = (1 << (exponent_size - 1)) - 1;
-    union FloatExtractor {
-        struct {
-            unsigned long long mantissa : mantissa_size;
-            unsigned exponent : exponent_size;
-            unsigned sign : 1;
-        };
-        double d;
-    } extractor;
-
-    extractor.d = value;
-    VERIFY(extractor.exponent != (1 << exponent_size) - 1);
-    // Exponent cannot be filled as than we must be NaN or infinity.
-
-    i32 real_exponent = extractor.exponent - exponent_bias;
-    if (real_exponent < 0) {
-        // |value| is less than 1, and we cannot be zero so if we are negative
-        // value must be greater and vice versa.
-        return bigint_is_negative ? CompareResult::DoubleGreaterThanBigInt : CompareResult::DoubleLessThanBigInt;
+    // Now both bigint and value have the same sign, so let's compare our magnitudes.
+    auto magnitudes_compare_result = m_unsigned_data.compare_to_double(fabs(value));
+
+    // If our mangnitudes are euqal, then we're equal.
+    if (magnitudes_compare_result == UnsignedBigInteger::CompareResult::DoubleEqualsBigInt)
+        return CompareResult::DoubleEqualsBigInt;
+
+    // If we're negative, revert the comparison result, otherwise return the same result.
+    if (value_is_negative) {
+        if (magnitudes_compare_result == UnsignedBigInteger::CompareResult::DoubleLessThanBigInt)
+            return CompareResult::DoubleGreaterThanBigInt;
+        else
+            return CompareResult::DoubleLessThanBigInt;
+    } else {
+        if (magnitudes_compare_result == UnsignedBigInteger::CompareResult::DoubleLessThanBigInt)
+            return CompareResult::DoubleLessThanBigInt;
+        else
+            return CompareResult::DoubleGreaterThanBigInt;
```

Comment:
We can just return `magnitude_compare_result` here

Reply:
That's a good point, but we have now SignedBigInteger::CompareResult and UnsignedBigInteger::CompareResult, should we unify them?

Reply:
Oh I missed that, I think we should keep just the Unsinged one

Reply:
Not very useful to have two different enums with the same values and even meaning
And since we include unsigned from signed you have it already

Reply:
Let me know if you think it's better to move 'CompareResult' out of the class scope.

Reply:
No I think keeping it in the class is fine, I think there a bunch more enums like that.